src/openforms/registrations/contrib/zgw_apis/typing.py

```diff
@@ -32,6 +32,7 @@ class RegistrationOptions(TypedDict):
     zgw_api_group: ZGWApiGroupConfig
     catalogue: NotRequired[CatalogueOption]
     case_type_identification: str
+    product: NotRequired[str]
```

Comment:
if you do `default=""` on the serializer, you can drop the `NotRequired` and that makes code less verbose

```suggestion
    product: str  # URL reference to a product in the case type
```

Reply:
I've changed this. The test code now does give warnings (in the editor, not while running the tests) that `product_url` (i've also renamed it) is missing. So to keep the editor UI clean, i've added `"product_url": ""` to some tests

Reply:
yes, good - another PR also enabled/enables type checking on the test modules to make those things fail CI

we want to keep types as simple as possible and that usually means that they're strict rather than lax :)